utils/validate-data-source.js

```diff
@@ -1,13 +1,30 @@
 const appRoot = require('app-root-path');
 
-const { validateFilePath } = appRoot.require('api/v1/db/json/fs-operations');
-
-const dbPath = 'tests/unit/mock-data.json';
+const { validateDBPath } = appRoot.require('api/v1/db/json/fs-operations');
+const { validateDBConnection } = appRoot.require('api/v1/db/oracledb/connection');
 
 /**
  * @summary Validate database configuration
+ * @param {string} dataSourceType data source type
  * @function
  */
-const validateDataSource = () => validateFilePath(dbPath);
+const validateDataSource = (dataSourceType) => {
+  const validationMethods = {
+    aws: null, // TODO: add AWS validation method
+    http: null, // TODO: add HTTP validation method
+    json: validateDBPath(),
+    oracledb: validateDBConnection(),
```

Comment:
I like the idea of mapping data type keys to validation methods but I'm thinking we should just map the keys to the functions themselves instead of executing the functions and mapping the return result. This way, if an api is configured to use a json db source, valid configurations won't be needed for any of the other methods. What do you think?

Reply:
Very good catch. Will change.